metals/src/main/scala/scala/meta/internal/parsing/ClassFinder.scala

```diff
@@ -4,32 +4,74 @@ import scala.annotation.tailrec
 
 import scala.meta.Defn
 import scala.meta.Pkg
+import scala.meta.Position
 import scala.meta.Tree
-import scala.meta.inputs.Position
 import scala.meta.internal.metals.MetalsEnrichments._
 import scala.meta.io.AbsolutePath
 
 import org.eclipse.{lsp4j => l}
 
+final case class ClassWithPos(
+    name: String,
+    pos: Position
+)
+
 class ClassFinder(trees: Trees) {
 
   def findClass(path: AbsolutePath, pos: l.Position): Option[String] =
-    findClass(path, pos, true)
+    findClass(path, pos, checkInnerClasses = true).filter(_.nonEmpty)
```

Comment:
Why do you need to check for `nonEmpty` ?

Reply:
If you pass a wrong position then you'll get just an empty string. I'll move this filter to the `private findClass`